subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -115,6 +145,10 @@ int z_impl_z_zsock_getaddrinfo_internal(const char *host, const char *service,
 		}
 	}
 
+	if (host == NULL) {
+		return getaddrinfo_null_host(port, hints, res);
```

Comment:
Why do we need a function for this? Just put the checks here as is.

Reply:
> Why do we need a function for this? Just put the checks here as is.

Well, getaddrinfo() has a lot of functionality, which we don't support yet. I'd expect it to grow 3-4 times, and then it will be really long, with a lot of awkwardly wrapped indent code, etc. Summing up, both z_impl_z_zsock_getaddrinfo_internal() and getaddrinfo_null_host() are supposed to grow noticeably, so splitting them like that makes sense.

Reply:
ok